build-system/tasks/serve.js

```diff
@@ -193,19 +226,19 @@ async function serve() {
 
 /**
  * Starts a webserver at the repository root to serve built files.
- * @param {boolean=} lazyBuild
+ * @param {ServerOptionsDef=} serverOptions
  */
-async function doServe(lazyBuild = false) {
+async function doServe(serverOptions = {}) {
   createCtrlcHandler('serve');
   const watchFunc = async () => {
     await restartServer();
   };
   watch(serverFiles).on('change', debounce(watchFunc, watchDebounceDelay));
-  if (argv.new_server) {
+  if (argv.new_server || serverOptions.new_server) {
     buildNewServer();
   }
-  await startServer({}, {lazyBuild}, {});
-  if (lazyBuild) {
+  await startServer({}, {lazyBuild: serverOptions.lazyBuild}, {});
```

Comment:
The third arg here (`ModeOptions`) will have to be `{esm: argv.esm}` for the server to actually serve the ESM binaries. We log the server mode here:

https://github.com/ampproject/amphtml/blob/9d4bf3117c3ab3c63e964407e3a546723189294a/build-system/server/app-utils.js#L59-L75

To locally test this, you can run the following:
```
gulp visual-diff --nobuild --esm --percy_disabled 
```

The server must print `Serving ESM JS`, and not `Serving minified JS`